src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -486,6 +489,133 @@ trait BronzeTransforms extends SparkSessionWrapper {
     }
   }
 
+  def prepClusterSnapshot(
+                           workspace: Workspace,
+                           pipelineSnapTS: TimeTypes,
+                           apiEnv: ApiEnv,
+                           database: Database,
+                           clusterSnapshotErrorsTarget: PipelineTable,
+                           config: Config,
+                           apiEndpointTempDir: String
+                         )(auditDF: DataFrame) : DataFrame = {
+
+    val clusterIDs = auditDF
+      .select(cluster_idFromAudit.alias("cluster_id"))
+      .filter(col("cluster_id").isNotNull)
+      .distinct()
+      .select("cluster_id").distinct().collect().map(x => x(0).toString)
+
+    var jobClusterIDs = Array[String]()
+    try {
+      val jobDF = getJobsBase(auditDF)
+      if (!jobDF.isEmpty) {
+        jobClusterIDs = jobDF.select('clusterId.alias("cluster_id")).distinct().collect().map(x => x(0).toString)
+      }
+    } catch {
+      case e: Exception => {
+        val message = s"No New records are their for Job Cluster in Audit Log: ${e.getMessage}"
+        logger.log(Level.WARN, message)
+      }
+    }
+
+    var clusterListIDs = Array[String]()
+
+    try {
+      val clusterListDF = workspace.getClustersDF(deriveApiTempDir(config.tempWorkingDir,apiEndpointTempDir,pipelineSnapTS))
+      if (!clusterListDF.isEmpty) {
+        clusterListIDs = clusterListDF.select("cluster_id").distinct().collect().map(x => x(0).toString)
+      }
+    } catch {
+      case e: Exception => {
+        val message = s"No New records are their for cluster/list api: ${e.getMessage}"
+        logger.log(Level.WARN, message)
+      }
+    }
+
+    val errorMsg = "No Data is present for cluster snapshot"
+
+    val allClusterIDs = (clusterIDs ++ jobClusterIDs ++ clusterListIDs).distinct
+
+    if (allClusterIDs.isEmpty) throw new NoNewDataException(s"No clusters could be found with new events. Please " +
+      s"validate your audit log input and clusters_snapshot_bronze tables to ensure data is flowing to them " +
+      s"properly. Skipping!", Level.ERROR)
+
+    logger.log(Level.INFO, "Calling APIv2, Number of cluster id:" + clusterIDs.length + " run id :" + apiEnv.runID)
```

Comment:
This message could be more user friendly, they don't really know what APIV2 is, so this could change to: s"Retrieving cluster information for ${clusterIDs.length}"

Reply:
Done